### PR TITLE
Use OpInfo for unsqueeze test

### PR DESCRIPTION
**Pull request by asi1024**
This PR is ready for #56774.

(cc: @mruberry, @emcastillo, @kmaehashi)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1633,6 +1633,26 @@ def sample_repeat_tile(op_info, device, dtype, requires_grad, **kwargs):
 
     return samples
 
+def sample_unsqueeze(op_info, device, dtype, requires_grad, **kwargs):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1636]: The extra samples are cool additions - nice work!

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1633,6 +1633,26 @@ def sample_repeat_tile(op_info, device, dtype, requires_grad, **kwargs):
...
+        ((3, 4, 5), -3),
+        ((), 0)
+    ]
+
+    samples = []
+    for shape, axis in shapes_and_axes:
+        tensor = make_tensor(shape, device, dtype, low=None, high=None,
+                             requires_grad=requires_grad)
+
+        for t in (tensor, tensor.T):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1651]: This part is a little tricky. Taking the transpose is clever, but it creates a dependency between two sample inputs.

For now I suggest simplifying and simply not taking the transpose and constructing a SampleInput from each tensor directly.

In the future we should automate more noncontiguous testing.

**Review by mruberry — COMMENTED** [review]
Looks good except for the dependency between sampleinputs; just ping me when that tweak is done and we'll get this landed, @asi1024!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (asi1024) [reply torch/testing/_internal/common_methods_invocations.py]: @mruberry Fixed! Thanks!

**Review by mruberry — APPROVED** [review]
Cool! Thanks @asi1024!

**Pull request closed without merging**